cpp/src/request.cpp

```diff
@@ -105,10 +105,11 @@ void Request::callback(void* request, ucs_status_t status)
                    _callback.target<void (*)(void)>());
   if (_callback) _callback(status, _callbackData);
 
-  ucp_request_free(request);
+  if (request != nullptr) ucp_request_free(request);
```

Comment:
TODO: request that UCX accepts `nullptr` in its `XXX_free` functions so that we don't have to litter user code with this kind of nonsense error checking. That would align the UCX APIs more closely with that of `free`.